### PR TITLE
fix 'Implement the file protocol restrictions' (close #1071)

### LINKED ISSUE #1071: Implement the file protocol restrictions
We need to get rid of cookie. ([details](https://html.spec.whatwg.org/#cookie-averse-document-object))
Also we need to define the `reqOrigin` value as full file url instead of generating from referer. ([details (paragraph 6)](https://tools.ietf.org/html/draft-abarth-origin-06#section-2))

### PR DESCRIPTION
**Pull request by LavrovArtem**
/cc @inikulin @miherlosev @VasilyStrelyaev

### CODE REVIEW

**Comment by LavrovArtem** [comment]
@testcafe-build-bot \retest

**Comment by LavrovArtem** [comment]
ping @inikulin

File: src/client/sandbox/code-instrumentation/properties/index.js [hunk 1]
```diff
@@ -142,8 +142,20 @@ export default class PropertyAccessorsInstrumentation extends SandboxBase {
 
             cookie: {
                 condition: domUtils.isDocument,
-                get:       () => this.cookieSandbox.getCookie(),
-                set:       (doc, cookie) => this.cookieSandbox.setCookie(doc, String(cookie), true)
+
+                get: () => {
+                    if (destLocation.getParsed().protocol !== 'file:')
```
Comment (inikulin) [inline src/client/sandbox/code-instrumentation/properties/index.js]: Why not encapsulate this logic in cookieSandbox?

File: src/request-pipeline/context.js [hunk 2]
```diff
@@ -145,8 +145,12 @@ export default class RequestPipelineContext {
...
 
             if (parsedReferer) {
-                this.dest.referer   = parsedReferer.dest.url;
-                this.dest.reqOrigin = urlUtils.getDomain(parsedReferer.dest);
+                this.dest.referer = parsedReferer.dest.url;
+
+                if (parsedReferer.dest.protocol === 'file:')
+                    this.dest.reqOrigin = parsedReferer.dest.url;
+                else
+                    this.dest.reqOrigin = urlUtils.getDomain(parsedReferer.dest);
```
Comment (inikulin) [inline src/request-pipeline/context.js]: You can use ternary operator here:
```js
this.dest.reqOrigin = parsedReferer.dest.protocol === 'file:' ?
                                 parsedReferer.dest.url :
                                 urlUtils.getDomain(parsedReferer.dest);
```

**Comment by LavrovArtem** [comment]
@testcafe-build-bot \retest

**Comment by LavrovArtem** [comment]
FPR

**Pull request merged**